python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1342,7 +1347,33 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
             ("gate_up_proj", "gate_proj", 0),
             ("gate_up_proj", "up_proj", 1),
         ]
-
+        if self.n_share_fusion_experts != 0:
+            weights_list = list(weights)
+            weights_dict = {k: v for (k, v) in weights_list}
+            suffix_list = [
+                'down_proj.weight', 'down_proj.weight_scale_inv',
+                'gate_proj.weight', 'gate_proj.weight_scale_inv',
+                'up_proj.weight', 'up_proj.weight_scale_inv'
+            ]
+            current_device = torch.cuda.current_device()
+            is_master = (current_device == 0)
+            for moe_layer in tqdm(range(self.config.num_hidden_layers),
+                                  desc=f"Cloning {self.n_share_fusion_experts} "
+                                  "replicas of shared expert into MoE",
+                                  disable=not is_master):
+                if moe_layer < self.config.first_k_dense_replace:
```

Comment:
nit: maybe we need to consider moe_layer_freq

Reply:
Yeah, I'll handle it.